front_end/webserver.py

```diff
@@ -1150,7 +1178,19 @@ def get(self, course, assignment, exercise, student_id, submission_id):
 class GetSubmissionsHandler(BaseUserHandler):
     def get(self, course, assignment, exercise, user_id):
         try:
-            submissions = content.get_submissions_basic(course, assignment, exercise, user_id)
+            assignment_details = content.get_assignment_details(course, assignment, True)
+            client_ip = get_client_ip_address(self.request)
+            user_info = self.get_user_info()
+
+            if assignment_details["allowed_ip_addresses"] and client_ip not in assignment_details[
+                "allowed_ip_addresses"] and not self.is_administrator():
```

Comment:
Please modify this so that instructors and assistants can also view the pages from any IP address.